packages/fxa-shared/subscriptions/configuration/helpers.ts

```diff
@@ -0,0 +1,164 @@
+    output = {
+      uiContent: {
+        ...output.uiContent,
+        ...localesConfig.uiContent,
+      },
+      urls: {
+        ...output.urls,
+        ...localesConfig.urls,
+      },
+      support: {
+        ...output.support,
+        ...localesConfig.support,
+      },
+    };
+  }
+
+  return output;
+};
+
+export const urlsFromProductConfig = (
+  plan: Plan,
+  userLocales: readonly string[] = [DEFAULT_LOCALE],
+  useFirestoreProductConfigs: boolean
+) => {
+  if (useFirestoreProductConfigs) {
+    return flattenProductConfig(plan, userLocales).urls;
+  } else {
+    const planMetadataConfig = productDetailsFromPlan(plan, userLocales);
+
+    return {
+      termsOfService:
+        planMetadataConfig.termsOfServiceURL ||
+        DEFAULT_PRODUCT_DETAILS.termsOfServiceURL!,
+      termsOfServiceDownload:
+        planMetadataConfig.termsOfServiceDownloadURL ||
+        DEFAULT_PRODUCT_DETAILS.termsOfServiceDownloadURL!,
+      privacyNotice:
+        planMetadataConfig.privacyNoticeURL ||
+        DEFAULT_PRODUCT_DETAILS.privacyNoticeURL!,
+      privacyNoticeDownload:
+        planMetadataConfig.privacyNoticeDownloadURL ||
+        DEFAULT_PRODUCT_DETAILS.privacyNoticeDownloadURL!,
+      cancellationSurvey: planMetadataConfig.cancellationSurveyURL,
+      download:
+        plan.product_metadata?.downloadURL || plan.plan_metadata?.downloadURL,
+    };
+  }
+};
+
+export const webIconConfigFromProductConfig = (
+  plan: Plan,
+  userLocales: readonly string[] = [DEFAULT_LOCALE],
+  useFirestoreProductConfigs: boolean
+) => {
+  if (useFirestoreProductConfigs) {
+    const planConfiguration = getPlanProductConfig(plan);
+    const urls = flattenProductConfig(plan, userLocales).urls;
+
+    return {
+      webIcon: urls.webIcon,
+      webIconBackground: planConfiguration.styles?.webIconBackground,
+    };
+  } else {
+    const combinedMetadata = metadataFromPlan(plan);
+
+    return {
+      webIcon: combinedMetadata.webIconURL,
+      webIconBackground: combinedMetadata.webIconBackground,
+    };
+  }
+};
+
+export const uiContentFromProductConfig = (
+  plan: Plan,
+  userLocales: readonly string[] = [DEFAULT_LOCALE],
+  useFirestoreProductConfigs: boolean
+) => {
+  if (useFirestoreProductConfigs) {
+    return flattenProductConfig(plan, userLocales).uiContent!;
+  } else {
+    const metadata = metadataFromPlan(plan);
+    const detailsFromPlan = productDetailsFromPlan(plan, userLocales);
+
+    return {
+      subtitle: detailsFromPlan.subtitle,
+      details: detailsFromPlan.details,
+      successActionButtonLabel: detailsFromPlan.successActionButtonLabel,
+      upgradeCTA: metadata.upgradeCTA,
+    };
+  }
+};
+
+export const productUpgradeFromProductConfig = (
+  plan: Plan,
+  useFirestoreProductConfigs: boolean
+) => {
+  if (useFirestoreProductConfigs) {
+    const planConfiguration = getPlanProductConfig(plan);
+    return {
+      productOrder:
+        (planConfiguration.productOrder &&
```

Comment:
Ah good catch. Since productOrder can be either number or undefined, I should be able to just get away with `productOrder !== undefined` right? Is there a better way to do it?